### ISSUE TITLE
ngcc cli doesn't seem to work with symlinked folders in node_modules (pnpm)

### ISSUE THREAD

**Issue by thekip** [angular/angular#40853]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/....
`@angular/compiler-cli`

### Description

When you have `node_modules` folder composed of symlinks (pnpm style) `ngcc` comand could not explore the targets and just skip processing. 

Though it's still working using Angular CLI during build/serve it's very inconvenient because blocking our's monorepo workflow which can build multiple apps in parallel. 

## 🔬 Minimal Reproduction
```
ng new ngcc-issue-pnpm --package-manager pnpm
cd ./ngcc-issue-pnpm
ngcc
```

Actual result is Nothing happened. `ngcc`.
Expected result `ngcc` should process packages in node_modules

You can run `ng serve` to prove that there are something to compile.

## 🌍  Your Environment

**Angular Version:**
<pre>

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/

Angular CLI: 11.2.0
Node: 14.15.4
OS: darwin x64

Angular: 11.2.0
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.0 (cli-only)
@angular-devkit/build-angular   0.1102.0
@angular-devkit/core            11.2.0 (cli-only)
@angular-devkit/schematics      11.2.0 (cli-only)
@schematics/angular             11.2.0 (cli-only)
@schematics/update              0.1102.0 (cli-only)
rxjs                            6.6.3
typescript                      4.1.5
</pre>

**PNPM Version:**
<pre>
λ pnpm --version
5.17.1
</pre>

**Comment by petebacondarwin** [comment]
This is basically a duplicate of https://github.com/angular/angular/issues/38023.

This should be solved by the ng-linker being developed to replace ngcc. Please follow along with the progress here: https://github.com/orgs/angular/projects/2
